### DISCUSSION TITLE
关于使用中可能遇到的一些问题

### DISCUSSION THREAD

**Discussion by liuyunfz** [liuyunfz/chaoxing_tool#3] [Q&A]
Q : 为什么密码无法输入，屏幕上没有显示密码？
A : 项目采用了与Linux类似的密码输入保护措施，即密码输入在屏幕上不可见，但却已准确的输入。与“*”符号替换密码异曲同工。同样如果你输入错了一个字符，那么你按下backspace键也是同样生效了的。所以并不是软件的问题，放心的输入你的密码吧。

Q : 为什么有的视频任务完成了，有的视频任务没完成？
A : ~~因为本项目目前对任务节点的完成情况判断并不完善，所以可能会导致程序误以为已经完成了，但实际上并未完成的情况。但类似的情况只会出现在 视频 类型的任务上，其他类型任务都会无条件重复完成，显然这也不是个好主意，只能期待后期的完善。~~ 以上问题已在 [第27次commit](https://github.com/liuyunfz/chaoxing_tool/commit/113b9d1430be32729dd7643c3fee6879feaea4df) 时修复了，如果仍出现了上述问题，请检查你使用的版本的发布时间。

Q : 我想要完成 测验/考试 该怎么办，好像并没有找到类似的功能啊？
A : 正如所说的，本程序并没有相关的功能，只能完成普通的任务类型（视频，直播，阅读等）。建议搭配使用油猴脚本——[超星学习通网课助手(考试专版)](https://greasyfork.org/zh-CN/scripts/418354-%E8%B6%85%E6%98%9F%E5%AD%A6%E4%B9%A0%E9%80%9A%E7%BD%91%E8%AF%BE%E5%8A%A9%E6%89%8B-%E8%80%83%E8%AF%95%E4%B8%93%E7%89%88)

Q : 我使用了这个程序会不会被老师监测到，会被封号吗？
A : 这个问题我其实不能很明确的答复，因为我并不清楚超星平台的监测机制，以及老师后台有什么数据。不过不要灰心，我已经在 3 门课上使用了此脚本，为期了一整个学期。期间老师也多次登录了后台，并没有看到相关的异常数据。所以还请放心使用——撑死胆大的，饿死胆小的。

Q :  我没有编程基础，没接触过Python，能用吗？
A : 考虑到不同的使用者，如果你有python基础，可以直接克隆仓库，然后在本地运行py文件，并根据自己需求适当修改代码。如果你没有python基础，也可以直接下载release（通过pyinstaller编译后，可以直接在windows上运行）使用，并不需要任何相关知识，如果你有一些特殊的需求也可以来找我，我会很乐意为你提供服务。

Q : 没有我想要的功能（打卡签到等）
A : 本项目基于作者本人遇到的内容进行开发，可能缺少功能的广泛性，如打卡签到虽然我有在其他仓库见到过，但因为未遇到过，所以并不清楚具体内容是什么，也就无从下手。欢迎与我联系（提个issue或给我Email），帮助我完善更多的功能。

---
以上是我想到的可能产生的疑问，如果你有其他的问题欢迎提issue或开启新的discussion。

**Comment by mezdosec** [comment]
脚本一键过任务点时候会报错，报错内容提示
课程读取完成，共有5个章节可一键完成
num: 0
error:list index out of range
